src/components/Tool/FixPhoneNumbers/Contact.tsx

```diff
@@ -1,46 +1,50 @@
 import NextLink from 'next/link';
-import React, { ReactElement, useMemo, useState } from 'react';
+import React, { ReactElement, useEffect, useState } from 'react';
 import styled from '@emotion/styled';
-import { mdiCheckboxMarkedCircle, mdiDelete, mdiLock } from '@mdi/js';
+import { mdiCheckboxMarkedCircle } from '@mdi/js';
 import { Icon } from '@mdi/react';
-import StarIcon from '@mui/icons-material/Star';
-import StarOutlineIcon from '@mui/icons-material/StarOutline';
 import {
   Avatar,
   Box,
   Button,
   Card,
   CardContent,
   CardHeader,
-  FormControl,
-  FormHelperText,
   Grid,
   Hidden,
   Link,
-  TextField,
-  Theme,
-  Tooltip,
   Typography,
 } from '@mui/material';
-import clsx from 'clsx';
-import { Formik } from 'formik';
-import { DateTime } from 'luxon';
+import { Formik, useFormikContext } from 'formik';
 import { useSnackbar } from 'notistack';
 import { useTranslation } from 'react-i18next';
 import { makeStyles } from 'tss-react/mui';
 import * as yup from 'yup';
 import { Confirmation } from 'src/components/common/Modal/Confirmation/Confirmation';
 import { useContactLinks } from 'src/hooks/useContactLinks';
 import useGetAppSettings from 'src/hooks/useGetAppSettings';
-import { useLocale } from 'src/hooks/useLocale';
-import { dateFormatShort } from 'src/lib/intlFormat';
-import { isEditableSource, sourceToStr } from 'src/utils/sourceHelper';
 import theme from '../../../theme';
+import { ContactPhoneNumbers } from './ContactPhoneNumbers';
 import { PersonInvalidNumberFragment } from './GetInvalidPhoneNumbers.generated';
 import PhoneValidationForm from './PhoneNumberValidationForm';
-import { useUpdatePhoneNumberMutation } from './UpdateInvalidPhoneNumbers.generated';
+import {
+  useUpdateInvalidPhoneNumbersMutation,
+  useUpdatePhoneNumberMutation,
+} from './UpdateInvalidPhoneNumbers.generated';
+
+const ImperativeSubmit: React.FC<{ submitAll: boolean }> = ({ submitAll }) => {
```

Comment:
Can you add a props interface like we do for most of our components? Sorry, when I gave you that code snippet, I was just writing out the MVP.

Reply:
I would probably also put it its own file and write at least one test for it.

In your test, make the component have a `<Formik>` form that contains a `<ImperativeSubmit submitForm>` (for test simplicity submit the form immediately). You can pass a `jest.fn()` to `onSubmit` and test that it is called.